### PR TITLE
Move forwarding_fee_proportional_millionths, forwarding_fee_base_msat to ChannelHandshakeConfig

### PR DESCRIPTION
**Pull request by matchacactus**
First step for https://github.com/lightningdevkit/rust-lightning/issues/216

### CODE REVIEW

**Comment by matchacactus** [comment]
The second commit broke `test_priv_forwarding_rejection` randomly. I have tried and failed to find out why, another 👀 for debug help?

**Comment by TheBlueMatt** [comment]
I'm somewhat confused as to the motivation for this PR - `ChannelHandshakeConfig` is all configuration options which are set at the beginning of a channel and then can not be changed afterwards - its all stuff in the "handshake". Instead, I think we should move things from `ChannelConfig` which cannot be updated after the handshake, specifically `announced_channel` (cannot be updated per lightning spec), `commit_upfront_shutdown_pubkey` (its literally about whether we include the pubkey in the handshake or not).

File: lightning/src/util/config.rs [hunk 1]
```diff
@@ -264,13 +264,10 @@ impl Default for ChannelConfig {
 }
 
 impl_writeable_tlv_based!(ChannelConfig, {
-	(0, forwarding_fee_proportional_millionths, required),
```
Comment (TheBlueMatt) [inline lightning/src/util/config.rs:267]: Note that we cannot remove `required` elements, as old clients will refuse to load data serialized by new clients (which we guarantee won't happen at least across a few releases). We instead need to write the default value.

**Comment by matchacactus** [comment]
> I'm somewhat confused as to the motivation for this PR - `ChannelHandshakeConfig` is all configuration options which are set at the beginning of a channel and then can not be changed afterwards - its all stuff in the "handshake". Instead, I think we should move things from `ChannelConfig` which cannot be updated after the handshake, specifically `announced_channel` (cannot be updated per lightning spec), `commit_upfront_shutdown_pubkey` (its literally about whether we include the pubkey in the handshake or not).

I misunderstood the issue - am I on the right track with https://github.com/lightningdevkit/rust-lightning/pull/1270? Closing this PR in favor of #1270

**Pull request closed without merging**